authorizations/authorization-common/src/main/java/org/apache/gravitino/authorization/common/PathBasedMetadataObject.java

```diff
@@ -31,7 +31,16 @@ public class PathBasedMetadataObject implements AuthorizationMetadataObject {
    */
   public enum Type implements AuthorizationMetadataObject.Type {
     /** A path is mapped the path of storages like HDFS, S3 etc. */
-    PATH(MetadataObject.Type.FILESET);
+    FILESET_PATH(MetadataObject.Type.FILESET),
+    /** A path is mapped the path of table storage like Hive. */
+    TABLE_PATH(MetadataObject.Type.TABLE),
+    /** A path is mapped the path of schema storage like Hive. */
+    SCHEMA_PATH(MetadataObject.Type.SCHEMA),
+    /** A path is mapped the path of cluster storage like Hive. */
+    CATALOG_PATH(MetadataObject.Type.CATALOG),
+    /** A path is mapped the path of all cluster storages like Hive. */
+    METALAKE_PATH(MetadataObject.Type.METALAKE);
```

Comment:
Can you help provide some rationale behind this change?

Reply:
Every entity may have locations. So we should have different types for different entities.

Reply:
I still don't get it. Maybe someone else do?
Maybe add some comments for people like me?

Reply:
I will. Maybe I should give you background. How does a Hive table store? First, we will store the metadata in the MySQL, And then we will store the data in the HDFS location. For Hive, there is a default location for the cluster level. We can also specify other locations for schema and table levels. So a path won't only be fileset.

Reply:
I think maybe it would be more appropriate to change `AuthorizationMetadataObject.Type` to `List` or `Set`.
Just like this:
```
public enum Type implements AuthorizationMetadataObject.Type {
    /** A path is mapped the path of storages like HDFS, S3 etc. */
    PATH(MetadataObject.Type.FILESET, MetadataObject.Type.TABLE, MetadataObject.Type.SCHEMA, MetadataObject.Type.CATALOG, MetadataObject.Type.METALAKE);
```

Reply:
Type has a method to get the metadata object type. So it's hard to use list or set.